adagucserverEC/CXMLGen.cpp

```diff
@@ -769,7 +778,8 @@ int CXMLGen::getStylesForLayer(WMSLayer *myWMSLayer) {
   if (myWMSLayer->dataSource->dLayerType == CConfigReaderLayerTypeCascaded) {
     return 0;
   }
-  if (myWMSLayer->dataSource->dLayerType == CConfigReaderLayerTypeLiveUpdate) {
+  if (myWMSLayer->dataSource->dLayerType == CConfigReaderLayerTypeLiveUpdate && myWMSLayer->dataSource->cfgLayer->DataPostProc.empty()) {
```

Comment:
Maybe make a generic function where you pass myWMSLayer->dataSource->cfgLayer as argument and then check if the solar terminator is configured or not.

Reply:
Great idea! This will make the code more readable/easy to understand.